core/app/models/concerns/spree/user_address_book.rb

```diff
@@ -0,0 +1,73 @@
+module Spree
+  module UserAddressBook
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :user_addresses, foreign_key: "user_id", class_name: "Spree::UserAddress"
+
+      has_many :addresses, through: :user_addresses
+
+      has_one :default_user_address, -> { where default: true}, foreign_key: "user_id", class_name: 'Spree::UserAddress'
+      has_one :default_address, through: :default_user_address, source: :address
+
+      # bill_address is only minimally used now, but we can't get rid of it without a major version release
```

Comment:
The credit card to address association was a precursor to this work and has been recent (https://github.com/solidusio/solidus/commit/aabbdf0726a7b6ac2ca27440d66531c3deda4408#diff-8e792ec275d2700cbf9bd5fd845df367). There's a rake task in that commit to do the back filling.

RE: UI changes -- This is very much driven by a UI Bonobos wants to add. This PR as is should not _require_ UI changes, but _enable_ them. At the moment User#bill_address is still stored on the User model. What's here shouldn't break existing behavior. If it does, please let me know so I can fix it.